packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,27 +8,43 @@ import { InitError, SupportError } from './errors/index.mjs'
  *
  * @internal
  * @virtual
+ * @template [T=HTMLElement]
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @protected
+   * @type {T}
+   */
+  $root
+
   /**
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
-   * @param {Element | null} [$root] - HTML element to use for component
+   * @param {T | null} [$root] - HTML element to use for component
    */
   constructor($root) {
     const childConstructor = /** @type {ChildClassConstructor} */ (
       this.constructor
     )
 
+    if (!($root instanceof HTMLElement)) {
+      throw new ElementError({
+        component: childConstructor,
+        identifier: 'Root element (`$root`)'
+      })
+    } else {
+      this.$root = $root
+    }
+
     childConstructor.checkSupport()
 
     this.checkInitialised($root)
 
     const moduleName = childConstructor.moduleName
 
     if (typeof moduleName === 'string') {
-      moduleName && $root?.setAttribute(`data-${moduleName}-init`, '')
+      moduleName && $root.setAttribute(`data-${moduleName}-init`, '')
```

Comment:
**suggestion** Using `this.$root` here will allow to drop the `?.` in the same way, but give us flexibility to refactor the storage of the root and/or the marking as initialised in a method if the constructor starts becoming too big.